### PR TITLE
Feature/time series props

### PR DESCRIPTION
**Pull request by frasercl**
Adds a field for time series step to props and state: `GlobalViewerSettings.time`. No UI controls this bit of state yet, but changing its value triggers the appropriate image to be loaded. Image loads triggered by only a time point change do not trigger a reset to 3D mode or recalculated transfer functions and mask alpha, as a completely new image would.

This addresses allen-cell-animated/volume-viewer#93, and may be enough to close it.

### CODE REVIEW

**Comment by frasercl** [comment]
TODO: bounds check against `image.imageInfo.times`?

File: src/aics-image-viewer/components/App/index.tsx [hunk 1]
```diff
@@ -135,9 +136,10 @@ const defaultProps: AppProps = {
 };
 
 /** A `useState` that also creates a getter function for breaking through closures */
-function useStateWithGetter<T>(initialState: T): [T, (value: T) => void, () => T] {
+function useStateWithGetter<T>(initialState: T | (() => T)): [T, (value: T) => void, () => T] {
   const [state, setState] = useState(initialState);
-  const stateRef = useRef(initialState);
+  const isSetterFunction = (val: T | (() => T)): val is () => T => typeof val === "function";
+  const stateRef = useRef(isSetterFunction(initialState) ? initialState() : initialState);
```
Comment (toloudis) [inline src/aics-image-viewer/components/App/index.tsx:142]: this useStateWithGetter function is a bit cryptic to me and lines like 141-142 hard to parse.  I don't know if there is a more readable formulation of line 141.  
Maybe a couple of more verbose comments saying what the intent is, and what is the difference between the `T` version and the `(()=>T)` version?

File: src/aics-image-viewer/components/App/index.tsx [hunk 2]
```diff
@@ -427,30 +429,23 @@ const App: React.FC<AppProps> = (props) => {
...
-
   const openImage = async (): Promise<void> => {
     const { fovPath, cellPath, baseUrl } = props;
     const path = viewerSettings.imageType === ImageType.fullField ? fovPath : cellPath;
     const fullUrl = `${baseUrl}${path}`;
 
-    if (path === currentlyLoadedImagePath) {
+    // If this is the same image at a different time, keep luts. If same image at same time, don't bother reloading.
+    const samePath = path === currentImageLoadSpec?.subpath;
+    if (samePath && viewerSettings.time === currentImageLoadSpec?.time) {
```
Comment (toloudis) [inline src/aics-image-viewer/components/App/index.tsx:448]: why is there a viewerSettings.time? Should it maybe be viewerSettings.loadSpec.time?  Is the idea of viewerSettings.time to store the value of the time slider?

Comment (frasercl) [reply src/aics-image-viewer/components/App/index.tsx:448]: Right, `viewerSettings.time` is the property which would eventually be controlled by the time slider. My thought is that `viewerSettings` is basically the subset of `View3d` and `Volume` properties which are controlled by the UI and so need to be duplicated in React state.

File: src/aics-image-viewer/components/App/index.tsx (hunk 1 above)
Comment (frasercl) [reply src/aics-image-viewer/components/App/index.tsx]: Yeah I got weird with this one. On a second look there's something far simpler I could have done to support what I was trying to support. (Which is the convention that, if the argument to `useState` is a function, that function is assumed to hold an expensive calculation of the initial value of that state, rather than itself being the initial value of state.)

File: src/aics-image-viewer/components/App/index.tsx (hunk 2 above)
Comment (toloudis) [reply src/aics-image-viewer/components/App/index.tsx:447]: ok the only concern here is that we also have a current load spec which also specifies a "current time".

**Review by toloudis — APPROVED** [review]

File: src/aics-image-viewer/components/App/index.tsx (hunk 2 above)
Comment (frasercl) [reply src/aics-image-viewer/components/App/index.tsx:447]: I replaced `currentlyLoadedImagePath` state (just a string) with the load spec used to fetch the current image, figuring that was the best way to support more closely examining the specific source of an image and detecting changes other than URL that still require reloading. There may be a better name than `currentImageLoadSpec` that makes this relatively limited purpose clearer?

Comment (toloudis) [reply src/aics-image-viewer/components/App/index.tsx:447]: IMO the naming is just fine - the situation with what do we load and from where is just more complicated now. And we are currently set up to only load a single t . The load spec is supposed to encapsulate "everything you need to load the volume data": multiresolution level, time etc.
